source/postprocess/crystal_preferred_orientation.cc

```diff
@@ -0,0 +1,833 @@
+                            string_stream_content_draw_volume_weighting << weighted_rotation_matrices[write_draw_volume_weighted_cpo[property_i].first][grain] << " ";
+                            break;
+
+                          case Output::EulerAngles:
+                            Assert(compute_raw_euler_angles == true,
+                                   ExcMessage("Internal error: writing out raw Euler angles, without them being computed."));
+                            string_stream_content_draw_volume_weighting << weighted_euler_angles[write_draw_volume_weighted_cpo[property_i].first][grain][0] << " "
+                                                                        <<  weighted_euler_angles[write_draw_volume_weighted_cpo[property_i].first][grain][1] << " "
+                                                                        <<  weighted_euler_angles[write_draw_volume_weighted_cpo[property_i].first][grain][2] << " ";
+                            break;
+
+                          default:
+                            Assert(false, ExcMessage("Internal error: raw CPO postprocess case not found."));
+                            break;
+                        }
+                    }
+                  string_stream_content_draw_volume_weighting << std::endl;
+                }
+            }
+        }
+
+      std::string filename_master = particle_file_prefix_master + "." + Utilities::int_to_string(dealii::Utilities::MPI::this_mpi_process (MPI_COMM_WORLD),4) + ".dat";
+      std::string filename_raw = particle_file_prefix_content_raw + "." + Utilities::int_to_string(dealii::Utilities::MPI::this_mpi_process (MPI_COMM_WORLD),4) + ".dat";
+      std::string filename_draw_volume_weighting = particle_file_prefix_content_draw_volume_weighting + "." + Utilities::int_to_string(dealii::Utilities::MPI::this_mpi_process (MPI_COMM_WORLD),4) + ".dat";
+
+      std::unique_ptr<std::string> file_contents_master = std::make_unique<std::string>(string_stream_master.str());
+      std::unique_ptr<std::string> file_contents_raw = std::make_unique<std::string>(string_stream_content_raw.str());
+      std::unique_ptr<std::string> file_contents_draw_volume_weighting = std::make_unique<std::string>(string_stream_content_draw_volume_weighting.str());
+
+      if (write_in_background_thread)
+        {
+          // Wait for all previous write operations to finish, should
+          // any be still active,
+          if (background_thread_master.joinable())
+            background_thread_master.join ();
+
+          // then continue with writing the master file
+          background_thread_master
+            = std::thread([ my_filename = std::move(filename_master),
+                            my_temporary_output_location = temporary_output_location,
+                            my_file_contents = std::move(file_contents_master)]()
+          {
+            writer (my_filename, my_temporary_output_location, *my_file_contents, false);
```

Comment:
Any reason compressing this file is never allowed?

Reply:
This is an index with not that much data relative to the other files. Compressing this will make reading it out a bit more annoying. It can be solved, but I didn't think it was worth it.